### ISSUE TITLE
Question about the memory

### ISSUE THREAD

**Issue by Arlo0321** [UCSB-NLP-Chang/DiffusionDisentanglement#3]
Hello Authors, I have a few questions here and would like to have them answered. 

If I understand correctly, this work requires optimizing only a very small number of λ parameters. Why is there such a large memory cost? Is there no need to freeze the LDM parameters during the optimization process?

**Comment by wuqiuche** [comment]
Hi @houjie8888,

Thank you for your interest in our work! We will answer them as follows:
(1) _Why is there such a large memory cost?_ The major cost is due to the back-propagation when optimizing $\lambda$. Although there is only a small number of $\lambda$, the back-propagation involves the whole U-net and thus needs large memory. In short, the small number of $\lambda$ contributes to a convenient representation of the disentangled attributes, and it is not for memory efficiency during optimization.
(2) _Is there no need to freeze the LDM parameters during the optimization process?_ In our work, the LDM parameters are fixed because we want to preserve its ability to synthesize novel images. This is particularly useful when transferring the learned $\lambda$ to novel images. If the parameters are not fixed, it might influence the transferring performance.

Hope these answer your questions!

**Comment by Arlo0321** [comment]
Thank you very much for your helpful and timely response! @wuqiuche 

There is still a little confusion. Would not a fixed LDM be like the pre-trained VGG used in the calculation of perceptual loss? In short, since the parameters of U-net do not need to be optimized, why do we need its gradient, can't it be done directly like using pre-trained VGG?

I would appreciate it if you could answer again.

**Comment by wuqiuche** [comment]
Hi @houjie8888,

Sure - Actually we use the diffusion model similar to using a pre-trained VGG, which means we disable gradients for the parameters of the U-net. And you are right - we don't need gradients for these parameters since we never update them.

However, We just have to leave the "torch.no_grad" unchecked for the U-net part, so that the loss can be back-propagated through the U-net, finally resulting in the gradient on $\lambda$. Otherwise we won't have gradient for $\lambda$.

Thank you for the questions, and hope this further clarifies our implementation!

**Comment by Arlo0321** [comment]
Thank you very much for your reply, it is a great help to me. @wuqiuche
